tooling/karaf-maven-plugin/src/main/java/org/apache/karaf/tooling/ArchiveMojo.java

```diff
@@ -115,11 +115,13 @@
     public void execute() throws MojoExecutionException, MojoFailureException {
         org.apache.maven.artifact.Artifact artifact = project.getArtifact();
         artifact.setFile(targetFile);
-        try {
-            if (project.getPackaging().equals("karaf-assembly") && !archiveTarGz && !archiveZip) {
-                throw new IllegalArgumentException("For karaf-assembly packaging, you have to specify at least one archive type (tar.gz or zip)");
-            }
 
+        // abort if there are no archives to be created
+        if (!archiveTarGz && !archiveZip) {
```

Comment:
I think it makes sense to keep the test on the `karaf-assembly` packaging as well. There are several users that use the `karaf:archive` goal with `pom` packaging. Thoughts ?

Reply:
While I don't have a problem bringing it back, it was an intentional change and it doesn't make any difference because this is what's being executed when packaging is not `karaf-assembly`:

https://github.com/apache/karaf/blob/a775677e8410b5667a09f8bb7eddac1834d22421/tooling/karaf-maven-plugin/src/main/java/org/apache/karaf/tooling/ArchiveMojo.java#L134-L141

…and since both `archiveZip` and `archiveTarGz` are `false`, no code is executed anyway and we can just return from the function at the beginning.

But, your call, if you want me to change it back.